Import community flow is broken

# Bug Report

`zQ3shn2Y2MFo81UjK4SCu7AqmL7eBoFqBNgyDHX7R1KfBrfQq`

User story:
1. I have a link to the community (or it's public key)
2. I want to spectate this community

    There're only 2 ways of doing this: 
    - click the link in chat - not available for me, I don't have any chats yet
    - import community - let's try

3. I go to `Communties` -> `Import Community` and paste the key/link

<img width="1312" alt="Screenshot 2024-07-10 at 17 23 12" src="https://github.com/status-im/status-desktop/assets/25482501/4cdb6ba5-7356-4b29-9105-2326b726f1cb">

4. I click `Import` and get this:

<img width="1312" alt="Screenshot 2024-07-10 at 17 23 14" src="https://github.com/status-im/status-desktop/assets/25482501/e6f74a6e-4c9e-4194-9152-58132c54c043">

Nothing else is happening after 🙁 

### Expected behavior

After clicking import, community is spectated. Just as if I clicked such link in a chat.

### Actual behavior

I get a "Cancel Membership Request" popup.

Indeed I can (sort of) reproduce but when I do, after the "Request to join..." popup, I always immediately get this crash:

```
DBG 2024-07-11 14:40:50.293+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=48216 file=core.nim:27 rpc_method=wakuext_checkPermissionsToJoinCommunity
DBG 2024-07-11 14:40:50.293+02:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=48223 file=threadpool.nim:56 messageType=AsyncCheckAllChannelsPermissionsTaskArg:ObjectType threadid=48223 task="{\"$type\":\"AsyncCheckAllChannelsPermissionsTaskArg:ObjectType\",\"communityId\":\"0x036dc0bcb627d55aa693096b447190eab3c5c57c117446ea44d93e97be0fe6ef46\",\"addresses\":[],\"vptr\":742936048,\"slot\":\"onAsyncCheckAllChannelsPermissionsDone\"}"
DBG 2024-07-11 14:40:50.294+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=48223 file=core.nim:27 rpc_method=wakuext_checkAllCommunityChannelsPermissions
DBG 2024-07-11 14:40:55.812+02:00 NewBE_callPrivateRPC                       topics="rpc" tid=48197 file=core.nim:27 rpc_method=wakuext_speedupArchivesImport
panic: runtime error: slice bounds out of range [5177:5000]

goroutine 638 [running]:
github.com/status-im/status-go/telemetry.(*Client).pushTelemetryRequest(0x1c0073c3fa8?, {0x1c0228a6000?, 0x6bc?, 0x0?})
	/home/ltinkl/git/status/status-desktop/vendor/status-go/telemetry/client.go:201 +0x719
github.com/status-im/status-go/telemetry.(*Client).Start.func2()
	/home/ltinkl/git/status/status-desktop/vendor/status-go/telemetry/client.go:141 +0x1ff
created by github.com/status-im/status-go/telemetry.(*Client).Start in goroutine 573
	/home/ltinkl/git/status/status-desktop/vendor/status-go/telemetry/client.go:126 +0xf6
SIGABRT: Abnormal termination.
```

--- a/src/app/modules/main/chat_section/module.nim
+++ b/src/app/modules/main/chat_section/module.nim
@@ -702,7 +702,6 @@ proc getChatItemFromChatDto(
   var viewersCanPostReactions = true
   var missingEncryptionKey = false
   if self.controller.isCommunity:
-    let communityChat = community.getCommunityChat(chatDto.id)
     # NOTE: workaround for new community chat, which is delivered in chatDto before the community will know about that
     if community.hasCommunityChat(chatDto.id):
       let communityChat = community.getCommunityChat(chatDto.id)

--- a/ui/app/mainui/Popups.qml
+++ b/ui/app/mainui/Popups.qml
@@ -277,8 +277,12 @@ QtObject {
         openPopup(importCommunitiesPopupComponent)
     }
 
-    function openCommunityIntroPopup(communityId, name, introMessage,
-                                     imageSrc, isInvitationPending) {
+    function openCommunityIntroPopup(communityId,
+                                    name,
+                                    introMessage,
+                                    imageSrc,
+                                    isInvitationPending
+                                    ) {
         openPopup(communityJoinDialogPopup,
                   {communityId: communityId,
                    communityName: name,
@@ -712,11 +716,11 @@ QtObject {
                 store: root.communitiesStore
                 onJoinCommunityRequested: {
                     close()
+                    communitiesStore.spectateCommunity(communityId)
                     openCommunityIntroPopup(communityId,
                                             communityDetails.name,
                                             communityDetails.introMessage,
                                             communityDetails.image,
-                                            communityDetails.access,
                                             root.rootStore.isMyCommunityRequestPending(communityId))
                 }
                 onClosed: destroy()

